routing.asciidoc

```diff
@@ -82,304 +99,201 @@ As Wei gets message from the escrow that Bob has deposited the 10 golden coins W
+The payment is extended in from payer to payee, but settled from payee to payer, as each of the intermediaries gain the solution to the outgoing condition, and use that (possibly augmenting it) to satisfy the incoming solution.
+
+Typically the payer rewards the intermediaries by sending slightly more than the payment amount, in order to allow the intermediaries to send out less with their outgoing payment than what they received from the incoming payment.
+The difference between these two payment values makes up the "forwarding fee" collected by the intermediary.
+
+=== Conditional Chained End to End Secure Payment === 
+
+With our final addition, we'll achieve "end to end security".
+By this we mean that: no intermediaries are able to "claim" the payment without first obtaining the solution from someone further down from them in the chain.
+Additionally, we also require that the amount the payer intended to send is fully received by the payee.
+Finally, we require that non of the intermediaries are able to "contaminate" the payment beyond giving incorrect directions to the party that directly follows them.
+In other words, the intermediary shouldn't be able to materially affect the propagation of the payment several hops away from it.
+
+== Hash Time Locked Contracts ==
+
+In this section, we'll construct a conditional chained end to end payment known as the HTLC. 
+At each step we'll add a new component, then examine it in light of our original definition to ensure all requirements and security properties are reached.
+
+First, the "condition". For an HTLC, the condition is typically the reveal of a hash pre-image that matches a particular hash.
+This hash is typically referred to as the "payment hash", with the pre-image being called the "payment pre-image".
+If the name didn't give too much away, for an HTLC, we'll use a _cryptographically secure_ hash function as one part of our condition.
+By using a cryptographic hash function, we ensure that it's infeasible for another party to "guess" the solution of our condition, it's easy for anyone to verify the solution, and there's only one "solution" to the condition.
+
+In order to implement the "refund" functionality, we rely on the "absolute time lock" functionality of Bitcoin script.
+
+With all that said, a basic Bitcoin script implementing a hash time-locked contract would look something like the following:
+```
+OP_SIZE 32 OP_EQUAL
+
+OP_IF
+    OP_HASH160 <ripemd(payHash)> OP_EQUALVERIFY
+    <receiver key>
+OP_ELSE
+   OP_CHECKLOCKTIMEVERIFY <timeout>
+   OP_DROP
+   <sender key>
+
+OP_CHECKSIGVERIFY
+```
+
+Alice can present this script to Bob in order to kick off the conditional payment.
+For the chained aspect, Alice needs to be able to communicate the proper payment details to each hop in the route.
+Recall that each hop will specify a forwarding fee rate, as well as other parameters that express their forwarding policy.
```

Comment:
lets split away the fee. I think I explain the fee in detail when I talk about the onion. Here (as above) it is kind of a distraction. It is correct but I think we don't loose anything if we leave that detail out

Reply:
I agree we dive into it later in the chapter, but IMO it's important that the user is _introduced_ to the concepts earlier to serve as a base which we'l use to build on later in the chapter.